qa/tasks/cephfs/test_volumes.py

```diff
@@ -676,6 +676,99 @@ def test_volume_create(self):
         # clean up
         self._fs_cmd("volume", "rm", volname, "--yes-i-really-mean-it")
 
+    def test_with_both_pool_names(self):
+        '''
+        Test that "ceph fs volume create" command accepts metadata pool name
+        and data pool name as arguments and uses these pools to create a new
+        volume.
+        '''
+        v = self._gen_vol_name()
+        meta = 'meta4521'
+        data = 'data4521'
+        self.run_ceph_cmd(f'osd pool create {meta}')
+        self.run_ceph_cmd(f'osd pool create {data}')
+        self.run_ceph_cmd(f'fs volume create {v} --data-pool {data} '
+                          f'--meta-pool {data}')
+        o = self.get_ceph_cmd_stdout('fs ls')
+        o = o.split('\n')
+        for i in o:
+            if f'name: {v},' in o:
+                self.assertIn(f'metadata pool: {meta}')
+                self.assertIn(f'datadata pool: [{data} ]')
+
+    def test_with_data_pool_name(self):
+        '''
+        Test that "ceph fs volume create" command runs successfully when data
+        pool name is passed and metadata pool name is not passed to the
+        command. Also test that the created volume uses the data pool passed by
+        user and creates a new metadata pool for the volume to use.
+        '''
+        v = self._gen_vol_name()
+        data = 'data4521'
+        self.run_ceph_cmd(f'osd pool create {data}')
+        self.run_ceph_cmd(f'fs volume create {v} --data-pool {data} ')
+        o = self.get_ceph_cmd_stdout('fs ls')
+        o = o.split('\n')
+        for i in o:
+            if f'name: {v},' in o:
+                self.assertIn(f'datadata pool: [{data} ]')
+
+    def test_with_meta_pool_name(self):
+        '''
+        Test that when only metadata pool name is passed to "ceph fs volume
+        create" command, the volume is created using the passed metadata pool
+        name and data pool for this volume is created automatically.
+        '''
+        v = self._gen_vol_name()
+        meta = 'meta4521'
+        self.run_ceph_cmd(f'osd pool create {meta}')
+        self.run_ceph_cmd(f'fs volume create {v} --meta-pool {meta}')
+        o = self.get_ceph_cmd_stdout('fs ls')
+        o = o.split('\n')
+        for i in o:
+            if f'name: {v},' in o:
+                self.assertIn(f'metadata pool: {meta}')
+
+    def test_with_nonempty_meta_pool_name(self):
```

Comment:
The pool emptiness check is done by the monitor volume create interface, yes?

Reply:
Yes

Reply:
Is this fine? Can I mark this thread as resolved?